tutorials/src/bin/basic-tutorial-5.rs

```diff
@@ -0,0 +1,377 @@
+#[cfg(feature = "tutorial5")]
+use std::process;
+#[cfg(feature = "tutorial5")]
+use std::os::raw::c_void;
+
+#[cfg(feature = "tutorial5")]
+extern crate glib;
+#[cfg(feature = "tutorial5")]
+use glib::translate::*;
+#[cfg(feature = "tutorial5")]
+use glib::*;
+
+#[cfg(feature = "tutorial5")]
+extern crate gdk;
+#[cfg(feature = "tutorial5")]
+use gdk::prelude::*;
+
+#[cfg(feature = "tutorial5")]
+extern crate gtk;
+#[cfg(feature = "tutorial5")]
+use gtk::*;
+
+#[cfg(feature = "tutorial5")]
+extern crate send_cell;
+#[cfg(feature = "tutorial5")]
+use send_cell::SendCell;
+
+#[cfg(feature = "tutorial5")]
+extern crate gstreamer as gst;
+#[cfg(feature = "tutorial5")]
+extern crate gstreamer_video as gst_video;
+#[cfg(feature = "tutorial5")]
+use gst_video::prelude::*;
+
+#[cfg(feature = "tutorial5")]
+// Extract tags from streams of @stype and add the info in the UI.
+fn add_streams_info(
+    playbin: &gst::Element,
+    textbufcell: &SendCell<gtk::TextBuffer>,
+    stype: String,
+) {
+    let textbuf = textbufcell.borrow();
+    let propname: &str = &format!("n-{}", stype);
+    let signame: &str = &format!("get-{}-tags", stype);
+
+    match playbin.get_property(propname).unwrap().get() {
+        Some(x) => for i in 0..x {
+            let tags = playbin.emit(signame, &[&i]).unwrap().unwrap();
+
+            if let Some(tags) = tags.get::<gst::TagList>() {
+                textbuf.insert_at_cursor(&format!("{} stream {}:\n ", stype, i));
+
+                if let Some(codec) = tags.get::<gst::tags::VideoCodec>() {
+                    textbuf.insert_at_cursor(&format!("    codec: {} \n", codec.get().unwrap()));
+                }
+
+                if let Some(codec) = tags.get::<gst::tags::AudioCodec>() {
+                    textbuf.insert_at_cursor(&format!("    codec: {} \n", codec.get().unwrap()));
+                }
+
+                if let Some(lang) = tags.get::<gst::tags::LanguageCode>() {
+                    textbuf.insert_at_cursor(&format!("    language: {} \n", lang.get().unwrap()));
+                }
+
+                if let Some(bitrate) = tags.get::<gst::tags::Bitrate>() {
+                    textbuf
+                        .insert_at_cursor(&format!("    bitrate: {} \n", bitrate.get().unwrap()));
+                }
+            }
+        },
+        None => {
+            eprintln!("Could not get {}!", propname);
+        }
+    }
+}
+
+#[cfg(feature = "tutorial5")]
+// Extract metadata from all the streams and write it to the text widget in the GUI
+fn analyze_streams(playbin: &gst::Element, textbufcell: &SendCell<gtk::TextBuffer>) {
+    let text = "";
+    {
```

Comment:
`text` is only used once, and only in this block here, not outside

Reply:
Done